modules/build/src/main/scala/scala/build/preprocessing/ScalaPreprocessor.scala

```diff
@@ -123,15 +128,15 @@ case object ScalaPreprocessor extends Preprocessor {
                 (reqs, scopedReqs, opts, updatedContent)
             }.getOrElse((BuildRequirements(), Nil, BuildOptions(), None))
           val s = PreprocessedSource.InMemory(
-            Left(v.source),
-            v.subPath,
+            originalPath = Left(v.source),
+            relPath = relPath,
             updatedContentOpt.getOrElse(content),
-            0,
-            Some(options),
-            Some(requirements),
+            ignoreLen = 0,
+            options = Some(options),
+            requirements = Some(requirements),
             scopedRequirements,
-            None,
-            v.scopePath
+            mainClassOpt = Some(className),
```

Comment:
Why do we need this? As it's a Scala file (not a script), we shouldn't have to come up with a main class name here. If users defined one, it should be found after compilation, doesn't it? (at least, that's how it should work in theory)

Reply:
The short answer is, if we don't provide a fallback main class name here, a `NoMainClassFoundError` gets thrown in[ `scala.build.Build`
](https://github.com/VirtusLab/scala-cli/blob/main/modules/build/src/main/scala/scala/build/Build.scala#L63)
We create [an inferred class name for Inputs.ScalaFile](https://github.com/VirtusLab/scala-cli/blob/main/modules/build/src/main/scala/scala/build/preprocessing/ScalaPreprocessor.scala#L75) in a similar fashion, which my intension was to stay consistent with.
In turn, the `mainClassOpt`  seems to be used as fallback for the found main class in [`scala.build.Build`](https://github.com/VirtusLab/scala-cli/blob/main/modules/build/src/main/scala/scala/build/Build.scala#L60)

It seems to me that if we don't need to set the `mainClassOpt` here (and rely on the compilation instead), then we probably shouldn't need the inferred class name for `Inputs.ScalaFile`, either?
Is passing the class name here meant solely for the Scala script wrapper?